### PR TITLE
pull request for #249

### LINKED ISSUE #246: Icon for (google) contacts not applying
Just installed this pack yesterday from F-Droid because I saw it in the Update-Stream, and it seems no icon is applied for the Stock Android Contacts App `com.google.android.contacts`.

I found this weird because it should be rather widespread, so I checked your `appfilter.xml`. I may or may not have found the culprit in that the activity seems to be defined multiple times:

[Line 465](https://github.com/dkanada/frost/blame/6094e597661e5b7a3563f35bf7ad94c466bd0a98/app/src/main/res/xml/appfilter.xml#L465)
[Line 471](https://github.com/dkanada/frost/blame/6094e597661e5b7a3563f35bf7ad94c466bd0a98/app/src/main/res/xml/appfilter.xml#L471)
And probably most importantly [Line 928](https://github.com/dkanada/frost/blame/6094e597661e5b7a3563f35bf7ad94c466bd0a98/app/src/main/res/xml/appfilter.xml#L928)

While 465 and 471 define the same icon `contacts` (I assume `drawable` refers to the filename?), 928  defines `people` instead, probably overwriting the previous definitions.

Unfortunately I don't know how to test for this.

### LINKED ISSUE #249: Icon Request
[20200130_080151.zip](https://github.com/dkanada/frost/files/4135823/20200130_080151.zip)

- [x] AMdroid
- [x] Fairphone FP3 Camera
- [x] comdirect core
- [x] comdirect photoTAN
- [x] comdirect Trading
- [x] ~~Google Contacts~~ (see #246)
- [x] ~~Deezer~~ (see #241, #242, #243)
- [x] GMX Mail
- [x] LeafPic Revived
- [x] Magic Earth (see also #245)
- [x] Musicolet
- [x] Todo Agenda
- [x] Twitch (see #74)
- [x] Vivaldi (beta)

### PR DESCRIPTION
**Pull request by mihawk90**
closes #249

# New icons:
* AMdroid
* comdirect core App
* comdirect photoTAN
* comdirect trading
* GMX Mail
* Magic Earth (see #245)
* Musicolet
* Twitch (see #74)
* Vivaldi
* Todo Agenda *(Alternative)* - similar to app's default theme of light (current day) and dark (following days)

# Reused icons:
* Fairphone FP3 camera - generic camera icon
* Google contacts - generic contacts icon (closes #246 - hopefully)
* LeafPic Revived - original LeafPic icon

### CODE REVIEW

**Comment by Donnnno** [comment]
Hi! 
I've noticed a couple of things:
- the comdirect icons have big rounded corners, I think it's better to use the square.svg template for that.
- the vivaldi icon: circle in squircle, why no make it just a circle ? 
- the Todo agenda icon is kinda dense now, I personally think the old one was a bit better.

Other than that is it a really good job :) 
That's my feedback for you.

**Comment by mihawk90** [comment]
Thanks for the feedback.

> the comdirect icons have big rounded corners, I think it's better to use the square.svg template for that.

Since that was the first icon I did I took the `crontosignswiss.svg` and modified that, which already comes with those corners (it's the same whitelabel app). But yeah, other then that no real reason, I was already thinking about re-doing that on the square template.

> the vivaldi icon: circle in squircle, why no make it just a circle ?
I was thinking about that initially but ultimately didn't go with it. The reason simply because for one thing I used the presskit SVG (as mentioned in the commit message) and for another IMO it looks better that way.
Original: 
image
It's not the first icon to go with a custom form either ;)

> the Todo agenda icon is kinda dense now, I personally think the old one was a bit better.
I guess ultimately that comes down to preference. I wasn't sure if I'd include it originally after I realised there already was one.
But either way, I traced the original App icon ± a few pixels. Not sure about being denser, comparing the two they seem quite similar in spacing.

**Comment by Donnnno** [comment]
- Ah good to hear, is this the first time designing icons btw?
Another thing that I want to question is the use of words in icons in this case comdirect, is it necessary or can you achieve the same function without the name in the icon? 

- Don't be to afraid to make something custom :) Vivaldi only works good because they have that red background, when it's monotone you don't have that background so there's nothing that adds much value to the design.

- The Todo icon is more personal preference i guess yea.
> Maybe there is a way of including both? 

That's possible we've done that with several icons in the past like firefox, adding **alt** in the name for example `todoagendaalt.svg`.

**Comment by mihawk90** [comment]
Yes, first time doing icons.

> Another thing that I want to question is the use of words in icons in this case comdirect, is it necessary or can you achieve the same function without the name in the icon?
I'm not sure if it's really possible otherwise. For the trading app possibly, for the others not sure.
The photoTAN is a generic icon used by the supplier of the whitelabel app (Vasco). Since photoTAN is a standardised TAN-generator format, multiple Apps use this icon (Commerzbank, comdirect, CrontoSign Swiss, Deutsche Bank - in a variation, ING DiBa, probably others), so it needs some sort of differentiation. I would have used an icon, but comdirect doesn't have a logo per-se, their lettering *is* their logo. So we'd have to come up with something completely new and I'm not creative enough to do that :P 
The core app literally just has the C on it, so there's not much else to use. And just the C would be too similar to the coinbase App. So the only other way would - again - be using something completely custom and I'm not good with that.

> Vivaldi only works good because they have that red background, when it's monotone you don't have that background so there's nothing that adds much value to the design.
I see what you're getting at, however IMO the overall appearance is kinda unique. Also working on that icon I noticed the circle isn't actually a perfect circle, it's slightly off, just hardly noticable. They also have a black/grey and white version in their presskit, which they also use for the Snapshot versions, and IMO it looks just as well.
IDK, I just like it :D Also I actually tried just doing the circle while I was trying to work around the even-odd they use, and it didn't look good for some reason. Popped the frame back on and it looked nicer somehow.

Either way in both cases, if someone comes up with a better icon it can be replaced anytime :)

> That's possible we've done that with several icons in the past like firefox, adding alt in the name for example `todoagendaalt.svg`.

Well I get that, but does that just pop up in the App then automatically?

**Comment by Donnnno** [comment]
> Yes, first time doing icons.

Woah nice work!

Thanks for the really clear explanation about the comdirect apps :)  If you like how it is now then it's good 

> IDK, I just like it :D Also I actually tried just doing the circle while I was trying to work around the even-odd they use, and it didn't look good for some reason. Popped the frame back on and it looked nicer somehow.

Ah okay, good to hear that let's keep it like that then!

> Well I get that, but does that just pop up in the App then automatically?

If it's supported by the launcher your using you have to change it manually in the launcher (lawnchair and nova support that as far as I know).

**Comment by mihawk90** [comment]
> If you like how it is now then it's good

Don't necessarily like it that much, but currently I don't see another way :D 

> If it's supported by the launcher your using you have to change it manually
Yeah I know about that, but I meant if the icon just becomes "magically available" without defining it somewhere, just because it exists. But I guess that's just how the app build works somehow then :D

File: app/src/main/res/xml/appfilter.xml [hunk 1]
```diff
@@ -468,7 +481,6 @@
     <item component="ComponentInfo{com.samsung.android.contacts/com.samsung.android.contacts.contactslist.PeopleActivity}" drawable="contacts" />
     <item component="ComponentInfo{com.sonyericsson.android.socialphonebook/com.sonyericsson.android.socialphonebook.LaunchActivity}" drawable="contacts" />
     <item component="ComponentInfo{com.sonyericsson.android.socialphonebook/com.sonyericsson.android.socialphonebook.DialerEntryActivity}" drawable="contacts" />
-    <item component="ComponentInfo{com.google.android.contacts/com.android.contacts.activities.PeopleActivity}" drawable="contacts" />
```
Comment (dkanada) [inline app/src/main/res/xml/appfilter.xml]: Just curious, why was this one removed?

File: app/src/main/res/xml/appfilter.xml [hunk 2]
```diff
@@ -924,9 +939,6 @@
     <!-- Google Clock -->
     <item component="ComponentInfo{com.google.android.deskclock/com.android.deskclock.DeskClock}" drawable="clock" />
 
-    <!-- Google Contacts -->
-    <item component="ComponentInfo{com.google.android.contacts/com.android.contacts.activities.PeopleActivity}" drawable="people" />
```
Comment (dkanada) [inline app/src/main/res/xml/appfilter.xml]: same here

File: app/src/main/res/xml/appfilter.xml (hunk 1 above)
Comment (dkanada) [reply app/src/main/res/xml/appfilter.xml]: I just noticed it was to fix the Google Contacts issue.

Comment (mihawk90) [reply app/src/main/res/xml/appfilter.xml]: As mentioned in the commit message of e8cc47163f05af14680574fce02010637c482d04 and dkanada/frost#246 

This occurence is a duplicate of line 465 (now 478)

File: app/src/main/res/xml/appfilter.xml (hunk 2 above)
Comment (mihawk90) [reply app/src/main/res/xml/appfilter.xml]: As mentioned in the commit message of e8cc47163f05af14680574fce02010637c482d04

This line references a drawable `people` that doesn't actually exist and is most likely the root cause for dkanada/frost#246
At least from my understanding this definition would overwrite the previous 2 (now 1) lines 465 (now 478) and 471 (now removed).

Comment (dkanada) [reply app/src/main/res/xml/appfilter.xml]: Ah I see now, you are indeed correct!

**Pull request merged**